gadget/gadget.go

```diff
@@ -835,6 +836,29 @@ func validateVolume(vol *Volume) error {
 		if err := validateVolumeStructure(&s, vol); err != nil {
 			return fmt.Errorf("invalid structure %v: %v", fmtIndexAndName(idx, s.Name), err)
 		}
+
+		start := *s.Offset
```

Comment:
If `start` and `end` is not used outside the `if` statement, please move these closer to the actual checks you have on line 855